### ISSUE TITLE
Sway fails to apply multiple pango tags, unless they're surrounded by spaces

### ISSUE THREAD

**Issue by ismay** [swaywm/sway#7698]
### Please fill out the following:
- **i3 PR:**
  - If this is new i3 functionality, please add a link to the i3 pull request.

It is not new i3 functionality.

- **Description:**
  - Please describe in plain English how Sway and i3's behaviors differ or what functionality Sway is lacking.

Sway renders pango markup in the binding mode indicator and the status_command output area differently from i3 when the pango markup is not surrounded by spaces.

The problem is that sway renders this incorrect: `"<small><b>text</b></small>"`, and this correct: `" <small><b>text</b></small> "`. The problem is that it seems to fail to apply the small tag, unless it is surrounded by spaces. i3 does not do this, it applies both tags, regardless of it being surrounded by spaces or not.

Below I've included examples of this for the mode indicator and status command output for both sway and i3. Note that only sway fails to apply the `small` tag, and only when it is not surrounded by spaces. i3 applies it in all cases, as expected.

### Binding mode

Sway binding mode indicator, with spaces:

```
set $wmmode " <small><b>wm: [l]ock [r]estart e[x]it</b></small> "
mode --pango_markup $wmmode {
  ...
}
```

![sway-mode-with-spaces](https://github.com/swaywm/sway/assets/7355199/baf91834-6687-4cbc-bb25-6a69c187ff87)

Sway binding mode indicator, without spaces:

```
set $wmmode "<small><b>wm: [l]ock [r]estart e[x]it</b></small>"
mode --pango_markup $wmmode {
  ...
}
```

![sway-mode-without-spaces](https://github.com/swaywm/sway/assets/7355199/db487789-ec85-4075-9e11-7bab94212518)

i3 binding mode indicator, with spaces:

```
set $wmmode " <small><b>wm: [l]ock [r]estart e[x]it</b></small> "
mode --pango_markup $wmmode {
  ...
}
```

![i3-mode-with-spaces](https://github.com/swaywm/sway/assets/7355199/36c90c57-3a66-40a8-a96f-2bd034d81bc1)

i3 binding mode indicator, without spaces:

```
set $wmmode "<small><b>wm: [l]ock [r]estart e[x]it</b></small>"
mode --pango_markup $wmmode {
  ...
}
```

![i3-mode-without-spaces](https://github.com/swaywm/sway/assets/7355199/850964d3-e0cd-48c5-94ca-7b5769196b42)

### Status_command output

Sway status_command output, with spaces:

```
time=$(date +"%d %b %H:%M")
echo " <small><b>${time}</b></small> "
```

![sway-swaybar-with-spaces](https://github.com/swaywm/sway/assets/7355199/c0e1a3d8-0e24-49af-abb9-c1462bb5cb76)

Sway status_command output, without spaces:

```
time=$(date +"%d %b %H:%M")
echo "<small><b>${time}</b></small>"
```

![sway-swaybar-without-spaces](https://github.com/swaywm/sway/assets/7355199/a9c2f83b-be99-486b-bd92-f60602b8ac84)

i3 status_command output, with spaces:

```
time=$(date +"%d %b %H:%M")
echo " <small><b>${time}</b></small> "
```

![i3-i3bar-with-spaces](https://github.com/swaywm/sway/assets/7355199/2cf02bff-b7c8-4067-b0bf-dbf34e2d1fbd)

i3 status_command output, without spaces:

```
time=$(date +"%d %b %H:%M")
echo "<small><b>${time}</b></small>"
```

![i3-i3bar-without-spaces](https://github.com/swaywm/sway/assets/7355199/4cb0d881-c4dc-48a0-80ff-053883a7afce)

**Comment by ismay** [comment]
To clarify, the screenshots were made with i3bar and swaybar. Both with i3blocks as status_command.

**Comment by rpigott** [comment]
I think the small tag doesn't apply unless there is some non-small text.

**Comment by ismay** [comment]
> I think the small tag doesn't apply unless there is some non-small text.

Just in sway you mean? Because i3 does apply it. Or is i3 in violation of how pango tags should be applied and is sway actually doing it correctly?

**Comment by rpigott** [comment]
In sway. I mean that the spaces in particular aren't relevant.

What font line is in your config?

**Comment by ismay** [comment]
> What font line is in your config?

I'll include the whole sway bar configuration, for context:

```
# theme
#######

set $blue #5294E2
set $dkblue #383C4A
set $brgray #D3DAE3
set $gray #666A74
set $red #FC4138
set $white #FFFFFF

#                       bd ---- bg ---- txt --- indic - childbd
client.focused          $gray   $gray   $gray   $blue   $blue
client.focused_inactive $gray   $gray   $gray   $gray   $gray
client.unfocused        $gray   $gray   $gray   $gray   $gray
client.urgent           $gray   $gray   $gray   $red    $red

# bar
#####

bar {
  font pango:Hack Nerd Font 12
  position top
  status_command i3blocks
  status_padding 8
  strip_workspace_numbers yes
  tray_output none
  workspace_min_width 50

  colors {
    background $dkblue
    statusline $brgray
    separator  $brgray

    # class            bd      bg      txt
    focused_workspace  $blue   $blue   $white
    active_workspace   $blue   $blue   $white
    inactive_workspace $dkblue $dkblue $brgray
    urgent_workspace   $red    $red    $white
    binding_mode       $red    $red    $white
  }
}
```

> In sway. I mean that the spaces in particular aren't relevant.

I see. So you mean that _any_ characters that are outside of the small tag would trigger the application of the small tag, right? So since i3 seems to do this differently, which wm is in the right here? Should sway emulate what i3 is doing (always apply the small tag), or is what i3 is doing a bug?

**Comment by rpigott** [comment]
Both should match whatever pango-view does I think, but it seems pretty
clear this is a sway bug.

**Comment by ismay** [comment]
Ok, yeah. Yeah that's what it seemed like to me as well. Let me know if there's anything I can do to help debug.
